fluss-flink/fluss-flink-common/src/main/java/com/alibaba/fluss/flink/catalog/FlinkTableFactory.java

```diff
@@ -200,17 +205,20 @@ public Set<ConfigOption<?>> optionalOptions() {
     }
 
     private static Configuration toFlussClientConfig(
-            ReadableConfig tableOptions, ReadableConfig flinkConfig) {
+            Map<String, String> tableOptions, ReadableConfig flinkConfig) {
         Configuration flussConfig = new Configuration();
         flussConfig.setString(
                 ConfigOptions.BOOTSTRAP_SERVERS.key(),
-                tableOptions.get(FlinkConnectorOptions.BOOTSTRAP_SERVERS));
+                tableOptions.get(FlinkConnectorOptions.BOOTSTRAP_SERVERS.key()));
+
         // forward all client configs
-        for (ConfigOption<?> option : FlinkConnectorOptions.CLIENT_OPTIONS) {
-            if (tableOptions.get(option) != null) {
-                flussConfig.setString(option.key(), tableOptions.get(option).toString());
-            }
-        }
+        tableOptions.forEach(
+                (key, value) -> {
+                    if (FlussConfigUtils.CLIENT_OPTIONS.containsKey(key)
+                            || key.startsWith(CLIENT_SECURITY_PREFIX)) {
```

Comment:
I think we can simplify to `if (key.startsWith(CLIENT_PREFIX))`. Otherwise, `FlussConfigUtils.CLIENT_OPTIONS.containsKey(key)` may have problems that can't forward keys if using a old client version that doesn't contain new `client.*` configurations  registered in the table.